core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -86,34 +63,27 @@ object Scheduler {
     * @return Snapshot of the current state, as well as Source which produces StateEvents and Sink which accepts SpecEvents
     */
   def asSourceAndSink(client: MesosClient, podRecordRepository: PodRecordRepository)(implicit mat: Materializer)
-    : (Future[StateSnapshot], Source[StateEventOrSnapshot, NotUsed], Sink[SchedulerCommand, Future[Done]]) = {
-    val flow = fromClient(client, podRecordRepository)
-    asSourceAndSink(flow)(mat)
+    : Future[(StateSnapshot, Source[StateEvent, NotUsed], Sink[SchedulerCommand, Future[Done]])] = {
+    fromClient(client, podRecordRepository).map {
+      case (snapshot, flow) =>
+        val (source, sink) = asSourceAndSink(flow)(mat)
+        (snapshot, source, sink)
+    }(CallerThreadExecutionContext.context)
   }
 
-  def asSourceAndSink(schedulerFlow: Flow[SchedulerCommand, StateOutput, NotUsed])(implicit mat: Materializer)
-    : (Future[StateSnapshot], Source[StateEventOrSnapshot, NotUsed], Sink[SchedulerCommand, Future[Done]]) = {
+  def asSourceAndSink[A, B](schedulerFlow: Flow[A, B, NotUsed])(
+      implicit mat: Materializer): (Source[B, NotUsed], Sink[A, Future[Done]]) = {
 
     val ((commandInputSubscriber, subscriberCompleted), commandInputSource) =
-      Source.asSubscriber[SchedulerCommand].watchTermination()(Keep.both).preMaterialize()
-
-    val firstStateOutput = commandInputSource
-      .via(schedulerFlow)
-      .runWith(Sink.head)
+      Source.asSubscriber[A].watchTermination()(Keep.both).preMaterialize()
 
-    val stateSnapshot = firstStateOutput.map { case (snapshot, _) => snapshot }(ExecutionContext.global)
+    val stateEvents = commandInputSource.via(schedulerFlow)
 
-    val stateEvents = Source.fromFuture(firstStateOutput).flatMapConcat { case (_, events) => events }
-
-    (
-      stateSnapshot,
-      stateEvents,
-      Sink.fromSubscriber(commandInputSubscriber).mapMaterializedValue(_ => subscriberCompleted))
+    (stateEvents, Sink.fromSubscriber(commandInputSubscriber).mapMaterializedValue(_ => subscriberCompleted))
   }
 
-  private[usi] def fromClient(
-      client: MesosClient,
-      podRecordRepository: PodRecordRepository): Flow[SchedulerCommand, StateOutput, NotUsed] = {
+  private[usi] def fromClient(client: MesosClient, podRecordRepository: PodRecordRepository)
```

Comment:
```suggestion
  def asFlow(client: MesosClient, podRecordRepository: PodRecordRepository)
```

This seems to be the replacement for the old `asFlow`.